src/main/java/com/google/api/codegen/viewmodel/OptionalArrayMethodView.java

```diff
@@ -71,6 +73,14 @@ public static Builder newBuilder() {
     return new AutoValue_OptionalArrayMethodView.Builder();
   }
 
+  public boolean isReturnEmpty() {
+    return !hasReturnValue();
+  }
+
+  public boolean hasRequestStreaming() {
```

Comment:
Maybe the terminology is different in Java vs dynamic languages? In Java, you provide one request object at a time to a `StreamObserver`.